spi/common/json-ld-spi/src/main/java/org/eclipse/edc/jsonld/spi/transformer/AbstractJsonLdTransformer.java

```diff
@@ -280,6 +283,60 @@ protected String transformString(JsonValue value, TransformerContext context) {
         }
     }
 
+    /**
+     * Transforms a JsonValue to a list of strings. If the value parameter is not of type JsonString, JsonObject or JsonArray,
+     * a problem is reported to the context and an empty list is returned.
+     *
+     * @param value   the value to transform
+     * @param context the transformer context
+     * @return the string list result
+     */
+    @Nullable
+    protected List<String> transformIdStringArray(JsonValue value, TransformerContext context) {
```

Comment:
this looks overcomplicated, `profile`, when not-null, is always a (non-empty) list, as verified by the validator, so it should be treated as such.
only the code from 315 to 324 can be used (please move it in the calling class, there's no reason to keep it here).
the "isEmpty" validation can be avoided as well

Reply:
done